File: server/Model/Editor.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany, ManyToMany } from "typeorm";
import { Fandom } from "./Fandom";
import { Edit } from "./Edit";
import { Style } from "./Style";
import { Comment } from "./Comment";
import { Bookmark } from "./Bookmark";

@Entity()
export class Editor {
    @PrimaryGeneratedColumn()
    editor_id: number

    @Column()
    editor_username: string

    @ManyToMany(() => Fandom, fandom => fandom.fandom_editors)
    editor_fandom: Fandom[];

    @Column()
    editor_name: string

    @Column()
    editor_expereience: number

    @OneToMany(() => Edit, edit => edit.edit_editor)
    editor_edits: Edit[]

    @Column()
    editor_img: string

    @Column("json")
    editor_platforms: string[]

    @ManyToMany(() => Style, style => style.style_editor)
    editor_styles: Style[]

    @Column()
    editor_app: string

    @OneToMany(() => Comment, comment => comment.comment_author)
    editor_comments: Comment[]

    @OneToMany(() => Bookmark, bookmark => bookmark.bookmark_owner)
    editor_bookmarks: Bookmark[]
}

File: server/src/main/java/laurriana/time2day/model/Editor.java
package laurriana.time2day.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Editor extends User implements UserDetails {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int editorId;

    private String editorUsername;

    @ManyToMany
    private List<Fandom> editorFandom = new ArrayList<>();

    private String editorName;
    private int editorExperience;

    @OneToMany
    @Transient
    private List<Edit> editorEdits = new ArrayList<>();

    private String editorImg;

    @ElementCollection
    private List<String> editorPlatforms = new ArrayList<>();

    @ManyToMany
    private List<Style> editorStyles = new ArrayList<>();

    private String editorApp;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return List.of();
    }

    @Override
    public String getUsername() {
        return editorUsername;
    }

}
